Display more metadata

pinoaffe opened the pull request:
Various changes to display more metadata, such as multiple artists, album years, etc, reflecting recent changes in libmpdel.

Also allows listing available genres and searching/filtering by genre.

If you want me to split this up into several PRs, just let me know, I figured this could probably all be tested in one go.

pinoaffe committed:
Add function to list all genres or a specific genre's songs

pinoaffe committed:
Add links to the genre list and to search by genre to the browser

DamienCassou reviewed (changes requested):
Thank you so much for your amazing work. I left a few comments. 
Some of them ask for more commits and I'm wondering if it would make sense to create several PRs.

Additionally, linting doesn't seem to pass.

pinoaffe commented on mpdel-song.el:
```diff
@@ -126,16 +126,33 @@ when the song changes.")
+(defun mpdel-song--join-indented (lst)
+  "Produce an indented list of the names of libmpdel-entities LST."
+  (string-join
+   (mapcar (lambda (str)
```
will do!

pinoaffe commented on mpdel-song.el:
```diff
@@ -126,16 +126,33 @@ when the song changes.")
+         (track (or (libmpdel--song-track song) ""))
+         (disc (or (libmpdel--song-disc song) ""))
+         (artist (mpdel-song--join-indented (libmpdel-artists song)))
+         (genre (mpdel-song--join-indented (libmpdel-genres song)))
```
will do!

pinoaffe committed:
Make the date column 12 chars wide when listing songs

Previously, a full yyyy-mm-dd date would be abbreviated, this allows a
full date to be displayed

pinoaffe committed:
List multiple artists per song when listing songs

Since libmpdel now supports reading several Artist tags per song,
display all artists instead of just the first Artist tag.

pinoaffe committed:
Display more metadata in libmpdel-song-mode

libmpdel recently added support for a lot more song metadata (such as
song genres, multiple song artists, and song performers) - this
lists said metadata if it is present

pinoaffe committed:
Add a date column when listing albums

Since libmpdel now supports storing an album's date, we added a column
to display an album's date when listing albums

pinoaffe committed:
List multiple artists per album when listing songs

Since libmpdel now supports reading several AlbumArtist tags per
album, display all artists instead of just the artist listed in
the first AlbumArtist tag.

pinoaffe committed:
Add default values when listing albums

Analogously to other `navigel-entity-to-columns` implementations,
default to the empty string when an album property is unbound

pinoaffe committed:
Propertize the strings displayed when listing albums

Analogously to other `navigel-entity-to-columns` implementations,
propertize the values displayed when listing albums

pinoaffe committed:
Use album format when listing all albums

Previously, when listing all albums, the tablist format was unset,
defaulting to the default format with only a "Name" column.  This
ensures that the already existing album format is used when listing
all albums

pinoaffe commented on mpdel-tablist.el:
```diff
@@ -142,6 +144,9 @@
 (navigel-method mpdel navigel-tablist-format ((_entity (eql current-playlist)))
   (mpdel-tablist--song-format))
 
+(navigel-method mpdel navigel-tablist-format ((_entity (eql albums)))
```
sure!

pinoaffe commented on mpdel-tablist.el:
```diff
@@ -105,7 +105,7 @@
           (list "Album" 30 t)
           (list "Disk" 4 t)
-          (list "Date" 5 t)
+          (list "Date" 12 t)
```
sure!

pinoaffe committed:
Make the linter happy

It seems like the linter now has different capitalization requirements
than previously, update the package headers to reflect this change

pinoaffe commented:
I've pushed a new version of the PR, with the commits split up, more elaboration in the commit messages, and an additional change that should hopefully make the linter happy - it at least did do so on my machine

DamienCassou reviewed (changes requested):
11 commits is overwhelming for me. Could you please split them into several PRs. This makes things much easier to manage for me.

DamienCassou commented on mpdel-tablist.el:
```diff
@@ -121,6 +130,9 @@
   (mpdel-tablist--song-format))
 
+(navigel-method mpdel navigel-tablist-format ((_entity libmpdel-genre))
+  (mpdel-tablist--song-format))
```
Thank you for teaching me my own code, I totally forgot about these details. Now I remember, thanks :-).
